haystack/preview/document_stores/in_memory/document_store.py

```diff
@@ -183,13 +183,16 @@ def write_documents(self, documents: List[Document], policy: DuplicatePolicy = D
         ):
             raise ValueError("Please provide a list of Documents.")
 
+        written_documents = len(documents)
         for document in documents:
             if policy != DuplicatePolicy.OVERWRITE and document.id in self.storage.keys():
                 if policy == DuplicatePolicy.FAIL:
                     raise DuplicateDocumentError(f"ID '{document.id}' already exists.")
                 if policy == DuplicatePolicy.SKIP:
                     logger.warning("ID '%s' already exists", document.id)
+                    written_documents -= 1
```

Comment:
This is not skipping. This seems like a bug. 🤔

Reply:
Oh I think a `continue` or an `else` is missing :eyes: Let me fix that